### PR TITLE
src/lib/crypto: uses native parser, implements byte <=> base64

### PR DESCRIPTION
**Pull request by Glumli**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ env:
 jobs:
   include:
     - stage: build staging
-      node_js: "7"
+      node_js: "6"
```
Comment (ibihim) [inline .travis.yml]: why so?

Comment (Glumli) [reply .travis.yml]: Ah fuck
Good catch. I was fighting with Travis and tested lots of stuff out

File: src/lib/crypto/index.js [hunk 2]
```diff
@@ -76,6 +58,53 @@ const convertBlobToArrayBufferView = blob =>
...
+    for (let i = 0, imod4 = 0; i < base64.length; i += 1, imod4 = i % 4) {
+        if (imod4 !== 0) {
+            bytes.push((
+                (base64map.indexOf(base64.charAt(i - 1)) & ((2 ** ((-2 * imod4) + 8)) - 1))
+                    << (imod4 * 2))
+                |
+                (base64map.indexOf(base64.charAt(i))
+                    >>> (6 - (imod4 * 2))));
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Glumli) [reply src/lib/crypto/index.js]: I'm currently working on the license comment

File: src/lib/crypto/index.js [hunk 3]
```diff
@@ -76,6 +58,53 @@ const convertBlobToArrayBufferView = blob =>
...
+/* eslint-disable no-bitwise */
+/**
+ * converts a base64 string to a Uint8Array
+ *
+ * @param {String} base64
+ * @returns {Uint8Array} - the bytes of the base64 string
+ */
+const convertBase64ToArrayBufferView = (base64) => {
+    // Remove non-base-64 characters
+    base64 = base64.replace(/[^A-Z0-9+/]/ig, '');
```
[the remark this answers is not in the document]
Comment (Glumli) [reply src/lib/crypto/index.js]: Thought so as well. But it also removes the = at the end

File: src/lib/crypto/index.js [hunk 4]
```diff
@@ -76,6 +58,53 @@ const convertBlobToArrayBufferView = blob =>
...
+/**
+ * converts the bytes to a base64 string
+ *
+ * @param {Uint8Array} bytes
+ * @returns {String} - the base64 representation of the bytes
+ */
+const convertArrayBufferViewToBase64 = (bytes) => {
+    const base64 = [];
+    for (let i = 0; i < bytes.length; i += 3) {
+        const triplet = (bytes[i] << 16) | (bytes[i + 1] << 8) | bytes[i + 2];
```
[the remark this answers is not in the document]
Comment (Glumli) [reply src/lib/crypto/index.js]: The problem is, that I didn't find any proper implementation of byte to base64, so "implementing" it on our own was the way I went for. Will take 30mins to research once more

**Pull request merged**